File: {{cookiecutter.project_slug}}/src/{{cookiecutter.project_slug}}/data_management/clean_data.py
import pandas as pd


def clean_data(data, data_info):
    """Clean data set.

    Information on data columns is stored in ``data_management/data_info.yaml``.

    Args:
        data (pandas.DataFrame): The data set.
        data_info (dict): Information on data set stored in data_info.yaml. The
            following keys can be accessed:
            - 'outcome': Name of dependent variable column in data
            - 'outcome_numerical': Name to be given to the numerical version of outcome
            - 'columns_to_drop': Names of columns that are dropped in data cleaning step
            - 'categorical_columns': Names of columns that are converted to categorical
            - 'column_rename_mapping': Old and new names of columns to be renamend,
                stored in a dictionary with design: {'old_name': 'new_name'}
            - 'url': URL to data set

    Returns:
        pandas.DataFrame: The cleaned data set.

    """
    data = data.drop(columns=data_info["columns_to_drop"])
    data = data.dropna()
    for cat_col in data_info["categorical_columns"]:
        data[cat_col] = data[cat_col].astype("category")
    data = data.rename(columns=data_info["column_rename_mapping"])

    numerical_outcome = pd.Categorical(data[data_info["outcome"]]).codes
    data[data_info["outcome_numerical"]] = numerical_outcome

    return data

File: {{cookiecutter.project_slug}}/src/{{cookiecutter.project_slug}}/data_management/clean_data.r
# ======================================================================================
# Functions
# ======================================================================================

clean_data <- function(data, data_info) {
  data[data_info[["columns_to_drop"]]] <- NULL
  data <- na.omit(data)
  for (cat_col in data_info[["categorical_columns"]]) {
    data[[cat_col]] <- as.factor(data[[cat_col]])
  }
  data <- plyr::rename(data, data_info[["column_rename_mapping"]])

  numerical_outcome <- ifelse(data[[data_info[["outcome"]]]] == "Yes", 1, 0)
  data[data_info[["outcome_numerical"]]] <- numerical_outcome

  return(data)
}

# ======================================================================================
# Get produces and depends_on from pytask
# ======================================================================================

args <- commandArgs(trailingOnly = TRUE)
path_to_yaml <- args[length(args)]
config <- yaml::yaml.load_file(path_to_yaml)

produces <- config[["produces"]]
depends_on <- config[["depends_on"]]

# ======================================================================================
# Main
# ======================================================================================

data_info <- yaml::yaml.load_file(depends_on[["data_info"]])
data <- read.csv(depends_on[["data"]])
data <- clean_data(data, data_info)
write.csv(data, file = produces, row.names = FALSE)
